### PR TITLE
Signals fix in test-service-load: send signals only when the client is not disconnected or disposed

### PR DESCRIPTION
**Pull request by Harivallabha-Rangarajan**
- in loadTestDataStore.ts, send signals only when runtime is not disconnected or disposed
- this should hopefully reduce the recent uptick in `submitSignalDisconnected` errors we have been seeing

### CODE REVIEW

File: packages/test/test-service-load/src/loadTestDataStore.ts [hunk 1]
```diff
@@ -513,7 +513,7 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
         const signalsGapMs = cycleMs / signalsPerCycle;
         let submittedSignals = 0;
         try {
-            while (submittedSignals < clientSignalsSendCount) {
+            while (submittedSignals < clientSignalsSendCount && this.runtime.connected && !this.runtime.disposed) {
                 // all the clients are sending signals;
                 // with signals, there is no particular need to have staggered writers and readers
                 this.runtime.submitSignal("generic-signal", true);
```
Comment (anthony-murphy) [inline packages/test/test-service-load/src/loadTestDataStore.ts:519]: what will not looping do? would it be better to just wrap the submit and submitted++ call, so we will basically just delay while disconnected?

Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/src/loadTestDataStore.ts:519]: That's valid, thanks! Making the change :")

Comment (anthony-murphy) [reply packages/test/test-service-load/src/loadTestDataStore.ts:519]: i think the disposed check should probably stay in the loop

Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/src/loadTestDataStore.ts]: Done, thanks!

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**